docs/debug/debian_kernel_rebuild.md

```diff
@@ -0,0 +1,148 @@
+### Recompile debian kernel to support APU2 GPIO
+
+`http://www.pcengines.ch/howto.htm#gpio`
+
+Get appropriate debian kernel source, e.g. from `linux-source-*` package.
+
+Copy source to host PC and compile with `gpio` module included:
+
+```
+CONFIG_GPIO_NCT5104D=m
+CONFIG_GPIO_SYSFS=y
+```
+
+```
+make menuconfig
+sudo apt-get install debhelper
+sudo apt-get install modutils
+sudo apt-get install kernel-package
+make-kpkg clean
+fakeroot make-kpkg  -j8 --initrd --revision=1.2.custom kernel_image
+```
+
+Copy .deb into sd card, boot and install:
+
+```
+dpkg-i linux-image-3.16.36_1.3.custom_amd64.deb
+```
+
+vi /etc/modules
+
+add:
+
+```
+gpio-nct5104d
+```
+
+```
+cd /sys/class/gpio
+echo 0 > export
+cd gpio0
+echo out > direction
+echo 1 > value
```

Comment:
@macpijan this would be true for pre 4.8 kernel, after 4.8 this method is supported but not recommended and will be removed in near future.